lib/tokenlist.cpp

```diff
@@ -223,13 +223,13 @@ void TokenList::addtoken(const Token *tok)
         return;
 
     if (mTokensFrontBack.back) {
-        mTokensFrontBack.back->insertToken(tok->str(), tok->originalName());
+        mTokensFrontBack.back->insertToken(tok->str(), tok->originalName(), tok->getMacroName());
```

Comment:
The whole macro name stuff looks super messy. It is not forwarded in all the other `addtoken()` functions and the forwarding is also missing all across the code.

I also wonder if we need an `Token::insertToken(const Token*)` to make sure all of the values are being transferred.